src/MyBudget.js

```diff
@@ -0,0 +1,213 @@
+    [theme.breakpoints.only("xs")]: {
+      marginLeft: theme.spacing(7),
+      marginTop: theme.spacing(7),
+    },
+    [theme.breakpoints.up("sm")]: {
+      marginLeft: theme.spacing(9),
+      marginTop: theme.spacing(8),
+    },
+  },
+});
+
+class MyBudget extends React.Component {
+  constructor(props) {
+    super(props);
+    const locallySavedState = LS.load(localStorageKey);
+    this.state = locallySavedState || defaultState;
+    this.state.currencyHandler = new CurrencyHandler();
+    this.state.currencyHandler.fetchAll(() => {
+      console.log("ERROR");
+    });
+    if (locallySavedState !== null) {
+      const locallySavedAccounts = locallySavedState.accounts;
+      const locallySavedTransactions = locallySavedState.transactions;
+      this.state.accounts = locallySavedAccounts.map((account) =>
+        AccountData.fromAccountObject(account)
+      );
+      this.state.transactions = locallySavedTransactions.map((transaction) =>
+        TransactionData.fromTransactionObject(transaction)
+      );
+    }
+  }
+
+  updateState(newState) {
+    this.setState(newState, () => {
+      LS.save(localStorageKey, this.state);
+    });
+  }
+
+  addAccount = (name, initialBalance, currencyId) => {
+    this.updateState({
+      accounts: [
+        ...this.state.accounts,
+        new AccountData(name, parseFloat(initialBalance), currencyId),
+      ],
+    });
+  };
+
+  updateAccount = (account) => {
+    let editedAccount = this.state.accounts.find(
+      (acc) => acc.id === account.id
+    );
+    editedAccount.editAccount({
+      name: account.name,
+      balance: parseFloat(account.balance) || 0,
+      currencyId: account.currencyId,
+    });
+    this.updateState({
+      accounts: this.state.accounts.map((currentAccount) =>
+        currentAccount.id === account.id ? editedAccount : currentAccount
+      ),
+    });
+  };
+
+  removeAccount = (uuid) => {
+    const filteredAccounts = this.state.accounts.filter(
+      (account) => account.id !== uuid
+    );
+    this.updateState({
+      accounts: filteredAccounts,
+    });
+  };
+
+  addTransaction = (amount, currencyId, accountId, date) => {
+    const transactions = this.state.transactions;
+    transactions.push(new TransactionData(amount, currencyId, accountId, date));
+    this.updateState({
+      transactions: transactions,
+    });
+  };
+
+  updateTransaction = (transaction) => {
+    this.updateState({
+      transactions: this.state.transactions.map((currentTransaction) =>
+        currentTransaction.id === transaction.id
+          ? transaction
+          : currentTransaction
+      ),
+    });
+  };
+
+  removeTransaction = (uuid) => {
+    const filteredTransactions = this.state.transactions.filter(
+      (transaction) => transaction.id !== uuid
+    );
+    this.updateState({
+      transactions: filteredTransactions,
+    });
+  };
+
+  toggleTransactionCommited = (transaction) => {
+    const account = this.state.accounts.find(
```

Comment:
This code contains duplication of logic.
What caused the duplication, why it is not a trivial task to remove it?
Could you think of a way to unify the two types of transaction commit?